client/state/ui/language/reducer.js

```diff
@@ -24,4 +25,34 @@ export const localeSlug = createReducer(
 	localeSlugSchema
 );
 
-export default combineReducers( { localeSlug } );
+/**
+ * Tracks the state of the ui locale
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ *
+ */
+export const isRtl = createReducer(
```

Comment:
I think that having a separate reducer for the `localeSlug` and `isRtl` values and composing them with `combineReducers` is overkill. `combineReducers` is great for combining several large independent reducers, but here we have one string and one boolean that both react only to `LOCALE_SET` action.

What about writing just one reducer without all the bloat?
```js
const initialState = { localeSlug: null, isRtl: null };
export default createReducer( initialState, {
  [ LOCALE_SET ]: ( state, { localeSlug } ) => ( {
    localeSlug,
    isRtl: getRtlForLocaleSlug( localeSlug ),
  } )
}, languageSchema );
```

The schema can be just one object with two properties, too.

Reply:
I don't think it's a good approach for future because if you need to add a completely separate reducer but under the `language` tree, you'll have to refactor it all. Or am I wrong?

Anyways, I'll do it as you want.

Reply:
We generally encourage composition since initial state gets quite complex, and state updates are harder to reason about. We're flexible of course, but composition for small pieces isn't unreasonable.

I'm sure @dmsnell can come up with a more eloquent argument.

Reply:
Note that using `createReducer` is always optional. It was originally created since folks really dislike switch statements :)

```jsx
export const isRtl = function( state = {}, action ) {
    if ( action.type === LOCALE_SET ) {
        // return state update.
    }
    return state;
}
isRtl.schema = isRtlSchema; 
```

Reply:
> I don't think it's a good approach for future because if you need to add a completely separate reducer but under the language tree, you'll have to refactor it all. Or am I wrong?

Yes, you're right -- but there's a counterargument with code name "YAGNI" 🤷‍♂️ 

I think that in this particular case, the `localeSlug` and `isRtl` properties belong naturally together, as they are always updated in sync with the `LOCALE_SET` action.

But if there's pushback, let's keep it as it is.

Reply:
Oh is it only 2 props? as long as the data makes sense together, I wouldn't block on that

Reply:
> Note that using createReducer is always optional.

:+1: on that. It made me realize that we don't need to persist the `ui.language` state to IndexedDB. In this case, the source of truth about the user's locale is somewhere else and a `LOCALE_SET` action is dispatched very very early during Calypso boot. That means that we can get rid of the schema, too?

Reply:
oh definitely if you don't need to persist, toss it out!

Reply:
I support both what you said @gwwar and what you said @jsnajdr 

Here's a fancy evaluator: would one of the properties ever change when the other wouldn't? Would we ever end up only re-rendering a small subset of the full re-render if only one changed?

We mash together related properties of API objects all the time; I'm okay with doing this. Still, I find that I'd rather code to maintainability as a first-principle and wait until it's a problem to make it a problem, meaning `combineReducers()` is really really nice.

This case to me is a gray area and with my general pessimism about JavaScript and performance I've been erring more often lately on the side of doing-less, meaning no `combineReducers()`

Either works, neither is clearly wrong, make a choice and it will all be okay.

---

> Note that using createReducer is always optional.

If it were me I'd say that the burden-of-proof required before using `createReducer()` should be through the roof. Actually, `switch` statements are super clear and incredibly fast and low-overhead. Probably the only complain is the additional `break` (insignificant) and the exception to normal scoping expectations (each `case` region is _not_ a separate scope - the entire `switch` block shares a scope).

We could remove a thousand cuts and make Calypso much faster. :+1:

Reply:
> each case region is not a separate scope - the entire switch block shares a scope

Yeah, this is the main trouble I run into for longer reducers, since I am bad a naming things. :) 💯 for doing less and keeping things simple when we can.